.github/workflows/rust.yml

```diff
@@ -254,9 +254,8 @@ jobs:
           cd ..
           rm -rf tmp # Delete the tmp folder
 
-      - name: Build test dependencies
-        if: matrix.directory == 'packages/engine'
-        run: cargo build -p hash_engine --manifest-path ${{ matrix.directory }}/Cargo.toml ${{ matrix.flags }}
+      - name: Build tests
```

Comment:
Why is this necessary? Can't we just run tests now?

Reply:
It's not needed, but when adding the python we need to build the cdylib to link them. Also when looking at the CI, it's nice, if the jobs are separated.

Reply:
> It's not needed, but when adding the python we need to build

I'm confused, is it needed or is it not?

If it's not needed then I don't think we should have it

Reply:
It's not needed currently.

Reply:
Sorry I just understood what you meant, when we add the Python changes that are currently WIP, we'll need to run a build as a separate step so we can link the cdylib in between building and running the tests. Gotcha!

Still think removing it for now was the right idea :) We might want to do the Python linking in build.rs or something later